torch/export/export_schema.py

```diff
@@ -0,0 +1,318 @@
+@dataclass
+class TensorArgument:
+    name: str   # identifier of the tensor, which must exist in graph's ivalues map
+
+# This is a SymInt Arugment used in the args of an node
+# We intentionally don't store the SymInt's value here, as the same SymInt argument can be used in multiple nodes
+# This field is an reference to the SymInt
+@dataclass
+class SymIntArgument:
+    name: str   # identifier of the symint, which must exist in graph's symint_values map
+
+#  Permissible return types for operators
+# !!! Notice: this assumes that a node can only return Tensor(s) and Symint(s), and not other int/float/bool types...
+# !!! What about .item()? Do we need to handle this?
+@dataclass
+class ReturnArgument:  # Union, ONLY EXACTLY ONE of the following fields can be set
+    as_tensor: TensorArgument = None
+    # as_tensors: List[TensorArgument] = None    # !!! ATM, no operator has return type as Tensor[], might need this latter?
+    as_symint: SymIntArgument = None
+
+
+# Permissible argument types for operators
+# !!! This is a Union struct, but there is no good python construct to model this
+@dataclass
+class Argument:  # Union, ONLY EXACTLY ONE of the following fields can be set
+    # None          # !!! This is used for nullable arguments, is this the right way to handle None?
+
+    as_tensor: TensorArgument = None
+    as_tensors: List[TensorArgument] = None   # Tensor[], used by aten.cat, and condition ops
+
+    as_symint: SymIntArgument = None         # Symint can be an argument, there are symint in native_function.yaml
+    as_symints: List[SymIntArgument] = None   # Symint[] can be an argement, there are symint[] in native_function.yaml
+
+    # as_scalar: Scalar = None      # !!! Looks like we don't need Scalar type during serialization,
+                                    # it will always be a concrete type, one of int, float, bool
+                                    # !!! Scalar is already an union type: Union[int, float, bool], check if serialization library can handle this
+    # List[Scalar], # !!! for Scalar[], used in native_function.yaml, but not used in canonical aten ops yet... Consider if we need this
+
+    as_bool: bool = None
+    as_bools: List[bool] = None     # for bool[]
+                                    # !!! There are use of bool[3] in canonical aten ops, consider if we can simplify this
+
+    as_int: int = None
+    as_ints: List[int] = None      # for int[]
+    as_float: float = None
+    as_floats: List[float] = None    # for float[]
+    as_str: str = None
+    # List[str],    # !!! There is no str[] in native_function.yaml. Consider if this is needed for expressiveness
+
+    # Graph,            # !!! Consider how to handle condition op, which need to pass in a graph for the branch
+    # List[Graph],      # !!! What about list of graphs? Do we need this?
+    as_gm: "GraphModule" = None     #  !!! ATM, torch.cond models branch as GraphModule
```

Comment:
@SherlockNoMad hmmm last time we reached a semi agreement that an Argument could be a Graph rather than a Graph which makes the graph structure much simpler. what changed your mind :)

Reply:
torch.cond models true_branch as GraphModule

Unless we change torch.cond, Argument needs to accept GraphModule

Reply:
do we really want to have 1:1 correspondence to torch.cond(), except the benefit of having consistency, I don't see much advantages. At least for graph representation of control flow, passing in a GraphModule is a compromise for FX rather than a intentional thing.